### DISCUSSION TITLE
Why does getMeasuringContext() values differ from getContext()

### DISCUSSION THREAD

**Discussion by klavsbuss** [fabricjs/fabric.js#7367] [Q&A] [answered]
I compared fabricjs `getMeasuringContext()` and Web API `getContext()` and run `measureText()` with same text style but they both returned different values. In fact fabricjs returned ~33,3333333333 times bigger values that made me think it's scaled differently? Can someone please explain why is it so?

Here is example: https://jsfiddle.net/vezgsL0b/2/

**Comment by asturur** [comment, marked as answer]
Hi @klavsbuss i did not manage to figure out quickly why you get different results.
getMeasuringContext will return you either the context of a canvas that we use for caching some operations, or a new canvas contenxt. It is supposed to be used just for internal text measurement but you should not rely on that for real measurements of an application.
We measure text at a fixed font size and then scale up or down at different font sizes.

one key difference is that the context you get from the canvas is retina scaled all the time, while the one you get for the caching layer is not.

**Reply by klavsbuss** [reply]
thanks, for insight!
